`inventory-report` fails with non-US datetime format settings

## Description

`inventory-report` fails with DateTime FormatException when computer date settings are non-US (i.e. datetime format is dd/mm/yyyy).

```
[07:55] [DEBUG] HTTP GET: https://dev.azure.com/mseng/vsplatdbgmdd/_apis/git/repositories/OpenDebugAD7/pushes?$top=1&api-version=7.1-preview.2
[07:55] [DEBUG] RESPONSE (OK): {"count":1,"value":[{"repository":{"id":"6af382c7-694c-45e5-8be5-0bf81d18464c","name":"OpenDebugAD7","url":"redactetd","project":{"id":"726cf03e-f8d6-4bd4-8aa6-b540f7c19b3d","name":"vsplatdbgmdd","description":"redacted","url":"redacted","state":"wellFormed","revision":293018731,"visibility":"organization","lastUpdateTime":"2018-04-02T04:30:30.17Z"},"size":6159880,"remoteUrl":"redacted","sshUrl":"redacted","webUrl":"readacted","isDisabled":false},"pushedBy":{"displayName":"redacted","url":"redacted","_links":{"avatar":{"href":"redacted"}},"id":"b628ec88-05ae-49ec-b175-32523249e866","uniqueName":"redacted","imageUrl":"redacted","descriptor":"redacted"},"pushId":1207944,"date":"2016-04-22T23:39:04.2658909Z","url":"redacted"}]}
[07:55] [ERROR] System.FormatException: String '04/22/2016 23:39:04' was not recognized as a valid DateTime.
   at System.DateTime.Parse(String s)
   at OctoshiftCLI.AdoApi.GetLastPushDate(String org, String teamProject, String repo) in C:\git\gh-gei\src\Octoshift\AdoApi.cs:line 68
   at OctoshiftCLI.AdoToGithub.ReposCsvGeneratorService.Generate(String adoPat) in C:\git\gh-gei\src\ado2gh\Services\ReposCsvGeneratorService.cs:line 36
   at OctoshiftCLI.AdoToGithub.Commands.InventoryReportCommand.Invoke(String adoOrg, String adoPat, Boolean verbose) in C:\git\gh-gei\src\ado2gh\Commands\InventoryReportCommand.cs:line 112
   at System.CommandLine.Invocation.CommandHandler.GetExitCodeAsync(Object value, InvocationContext context)
   at System.CommandLine.Invocation.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass23_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass27_0.<<UseVersionOption>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass25_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__24_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseParseDirective>b__0>d.MoveNext()
```

## Reproduction Steps

--- a/RELEASENOTES.md
+++ b/RELEASENOTES.md
@@ -1,2 +1,2 @@
 - fixed a bug where `generate-script` would not properly handle Team Projects with spaces (or other invalid for github chars) in the name
-- fixed bug where `inventory-report` would fail if your computers datetime format settings were dd/mm/yyyy
+- fixed bug where `inventory-report` would fail if your computer's datetime format settings were dd/mm/yyyy

--- a/RELEASENOTES.md
+++ b/RELEASENOTES.md
@@ -1,1 +1,2 @@
 - fixed a bug where `generate-script` would not properly handle Team Projects with spaces (or other invalid for github chars) in the name
+- fixed bug where `inventory-report` would fail if your computers datetime format settings were dd/mm/yyyy

--- a/src/Octoshift/AdoApi.cs
+++ b/src/Octoshift/AdoApi.cs
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
 using System.Linq;
 using System.Net;
@@ -63,20 +64,20 @@ namespace OctoshiftCLI
             var response = await _client.GetAsync(url);
 
             var data = JObject.Parse(response);
-            var pushDate = data.TryGetValue("value", out var dataValue) && dataValue.Any() ? (string)dataValue.First()["date"] : DateTime.MinValue.ToString();
+            var pushDate = data.TryGetValue("value", out var dataValue) && dataValue.Any() ? (DateTime)dataValue.First()["date"] : DateTime.MinValue;
 
-            return DateTime.Parse(pushDate);
+            return pushDate.Date;
         }
 
         public virtual async Task<int> GetCommitCountSince(string org, string teamProject, string repo, DateTime fromDate)
         {
-            var url = $"{_adoBaseUrl}/{org}/{teamProject}/_apis/git/repositories/{repo}/commits?searchCriteria.fromDate={fromDate.ToShortDateString()}&api-version=7.1-preview.1";
+            var url = $"{_adoBaseUrl}/{org}/{teamProject}/_apis/git/repositories/{repo}/commits?searchCriteria.fromDate={fromDate.ToString("d", CultureInfo.InvariantCulture)}&api-version=7.1-preview.1";
             return await _client.GetCountUsingSkip(url);
         }
 
         public virtual async Task<IEnumerable<string>> GetPushersSince(string org, string teamProject, string repo, DateTime fromDate)
         {
-            var url = $"{_adoBaseUrl}/{org}/{teamProject}/_apis/git/repositories/{repo}/pushes?searchCriteria.fromDate={fromDate.ToShortDateString()}&api-version=7.1-preview.1";
+            var url = $"{_adoBaseUrl}/{org}/{teamProject}/_apis/git/repositories/{repo}/pushes?searchCriteria.fromDate={fromDate.ToString("d", CultureInfo.InvariantCulture)}&api-version=7.1-preview.1";
             var response = await _client.GetWithPagingTopSkipAsync(url, x => $"{x["pushedBy"]["displayName"]} ({x["pushedBy"]["uniqueName"]})");
 
             return response;

--- a/src/OctoshiftCLI.Tests/AdoApiTests.cs
+++ b/src/OctoshiftCLI.Tests/AdoApiTests.cs
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
 using System.Linq;
 using System.Net;
@@ -532,10 +533,34 @@ namespace OctoshiftCLI.Tests
         }
 
         [Fact]
+        public async Task GetLastPushDate_Should_Be_Locale_Independent()
+        {
+            var endpoint = $"https://dev.azure.com/{ADO_ORG}/{ADO_TEAM_PROJECT}/_apis/git/repositories/{ADO_REPO}/pushes?$top=1&api-version=7.1-preview.2";
+            var expectedDate = new DateTime(2016, 4, 22);
+
+            var response = new
+            {
+                value = new[]
+                {
+                    new { date = "2016-04-22T23:39:04.2658909Z" }
+                }
+            };
+
+            _mockAdoClient.Setup(x => x.GetAsync(endpoint)).ReturnsAsync(response.ToJson());
+
+            CultureInfo.CurrentCulture = new CultureInfo("en-AT"); // Austrian culture has reversed datetime format
+
+            var result = await sut.GetLastPushDate(ADO_ORG, ADO_TEAM_PROJECT, ADO_REPO);
+
+            result.Should().Be(expectedDate);
+        }
+
+        [Fact]
         public async Task GetCommitCountSince_Should_Return_Commit_Count()
         {
             var fromDate = new DateTime(2022, 2, 14);
-            var endpoint = $"https://dev.azure.com/{ADO_ORG}/{ADO_TEAM_PROJECT}/_apis/git/repositories/{ADO_REPO}/commits?searchCriteria.fromDate={fromDate.ToShortDateString()}&api-version=7.1-preview.1";
+            var fromDateIso = "02/14/2022";
+            var endpoint = $"https://dev.azure.com/{ADO_ORG}/{ADO_TEAM_PROJECT}/_apis/git/repositories/{ADO_REPO}/commits?searchCriteria.fromDate={fromDateIso}&api-version=7.1-preview.1";
             var expectedCount = 12;
 
             _mockAdoClient.Setup(x => x.GetCountUsingSkip(endpoint)).ReturnsAsync(expectedCount);
@@ -546,10 +571,28 @@ namespace OctoshiftCLI.Tests
         }
 
         [Fact]
+        public async Task GetCommitCountSince_Should_Be_Locale_Independent()
+        {
+            var fromDate = new DateTime(2022, 2, 14);
+            var fromDateIso = "02/14/2022";
+            var endpoint = $"https://dev.azure.com/{ADO_ORG}/{ADO_TEAM_PROJECT}/_apis/git/repositories/{ADO_REPO}/commits?searchCriteria.fromDate={fromDateIso}&api-version=7.1-preview.1";
+            var expectedCount = 12;
+
+            _mockAdoClient.Setup(x => x.GetCountUsingSkip(endpoint)).ReturnsAsync(expectedCount);
+
+            CultureInfo.CurrentCulture = new CultureInfo("en-AT"); // Austrian culture has reversed datetime format
+
+            var result = await sut.GetCommitCountSince(ADO_ORG, ADO_TEAM_PROJECT, ADO_REPO, fromDate);
+
+            result.Should().Be(expectedCount);
+        }
+
+        [Fact]
         public async Task GetPushersSince_Should_Return_Pushers()
         {
             var fromDate = new DateTime(2022, 2, 14);
-            var endpoint = $"https://dev.azure.com/{ADO_ORG}/{ADO_TEAM_PROJECT}/_apis/git/repositories/{ADO_REPO}/pushes?searchCriteria.fromDate={fromDate.ToShortDateString()}&api-version=7.1-preview.1";
+            var fromDateIso = "02/14/2022";
+            var endpoint = $"https://dev.azure.com/{ADO_ORG}/{ADO_TEAM_PROJECT}/_apis/git/repositories/{ADO_REPO}/pushes?searchCriteria.fromDate={fromDateIso}&api-version=7.1-preview.1";
             var pusher1DisplayName = "Dylan";
             var pusher1UniqueName = "dsmith";
             var pusher2DisplayName = "Tom";
@@ -571,6 +614,42 @@ namespace OctoshiftCLI.Tests
 
             _mockAdoClient.Setup(x => x.GetWithPagingTopSkipAsync(endpoint, It.IsAny<Func<JToken, string>>()))
                 .ReturnsAsync((string url, Func<JToken, string> selector) => responseArray.Select(selector));
+
+            var result = await sut.GetPushersSince(ADO_ORG, ADO_TEAM_PROJECT, ADO_REPO, fromDate);
+
+            result.First().Should().Be("Dylan (dsmith)");
+            result.Last().Should().Be("Tom (tcruise)");
+        }
+
+        [Fact]
+        public async Task GetPushersSince_Should_Be_Locale_Independent()
+        {
+            var fromDate = new DateTime(2022, 2, 14);
+            var fromDateIso = "02/14/2022";
+            var endpoint = $"https://dev.azure.com/{ADO_ORG}/{ADO_TEAM_PROJECT}/_apis/git/repositories/{ADO_REPO}/pushes?searchCriteria.fromDate={fromDateIso}&api-version=7.1-preview.1";
+            var pusher1DisplayName = "Dylan";
+            var pusher1UniqueName = "dsmith";
+            var pusher2DisplayName = "Tom";
+            var pusher2UniqueName = "tcruise";
+
+            var response = new[]
+            {
+                new
+                {
+                    pushedBy = new { displayName = pusher1DisplayName, uniqueName = pusher1UniqueName }
+                },
+                new
+                {
+                    pushedBy = new { displayName = pusher2DisplayName, uniqueName = pusher2UniqueName }
+                }
+            }.ToJson();
+
+            var responseArray = JArray.Parse(response);
+
+            _mockAdoClient.Setup(x => x.GetWithPagingTopSkipAsync(endpoint, It.IsAny<Func<JToken, string>>()))
+                .ReturnsAsync((string url, Func<JToken, string> selector) => responseArray.Select(selector));
+
+            CultureInfo.CurrentCulture = new CultureInfo("en-AT"); // Austrian culture has reversed datetime format
 
             var result = await sut.GetPushersSince(ADO_ORG, ADO_TEAM_PROJECT, ADO_REPO, fromDate);
 

